### PR TITLE
Add renderer for hydra

### PR DESCRIPTION
**Pull request by emptyflash**
The biggest change here is duplicating `minimalHeader` and `fragFooter` and tweaking the GLES version (regl doesn't support 300) and tweaking things to work in that version.

Basic API example:
```
const {sculptToHydraRenderer} = await import("https://local.emptyfla.sh/shader-park-core.esm.js")

sculptToHydraRenderer(()=>{
  sphere(nsin(time))
}).out(o0)
```

Full demo: https://hydra.ojack.xyz/?sketch_id=Lovo7ySixRsokMQI

### CODE REVIEW

File: glsl/glsl-lib.js [hunk 1]
```diff
@@ -504,7 +512,7 @@ vec4 sphericalDistribution( vec3 p, float n )
 
     float k = max(2.0, floor( log(n * PI * sqrt(5.0) * (1.0 - cosTheta*cosTheta))/ log(PHI+1.0)));
     float Fk = pow(PHI, k)/sqrt(5.0);
-    vec2 F = vec2( round(Fk), round(Fk * PHI) ); // k, k+1
+    vec2 F = vec2( floor(Fk + 0.5), floor(Fk * PHI + 0.5) ); // k, k+1
```
Comment (PWhiddy) [inline glsl/glsl-lib.js:515]: Was this for webgl1 compatibility? We did migrate shaderpark to webgl2 a while ago.

**Review by PWhiddy — APPROVED** [review]
At first glance everything looks good! 
Hopefully the amount of code duplication between targets wasn't too annoying, we wanted to make sure that too much structure didn't make it difficult to implement new targets. 

Generally we'd prefer to build the dist packages ourselves, but in this case it should be fine. 

And don't mind the GA tests not passing, we need to fix those. 

Thanks again!
I'll let @torinmb take a took too and we should be good to merge.

**Review by emptyflash — COMMENTED** [review]
@PWhiddy I can always undo the dist changes, figured they might be done adhoc since they're committed

File: glsl/glsl-lib.js (hunk 1 above)
Comment (PWhiddy) [reply glsl/glsl-lib.js:515]: I see. I suspect it's possible there may be other places in the library that could have issues with webg1 besides this line. I'm happy to merge now and just fix them later if they show up.

**Pull request merged**